### PR TITLE
[SQL] Create SQL cog

### PR DESCRIPTION
**Pull request by OofChair**
:) New cog

### CODE REVIEW

File: sql/__init__.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+from redbot.core.bot import Red
+
+from .sql import SQL
+
+with open(Path(__file__).parent / "info.json") as fp:
+    __red_end_user_data_statement__ = json.load(fp)["end_user_data_statement"]
+
+
+async def setup(bot: Red) -> None:
+    bot.add_cog(SQL(bot))
```
Comment (Just-Jojo) [inline sql/__init__.py:13]: Dunno why this is async. (Optional)

File: sql/sql.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+    @mysql.group()
+    async def delete(self, ctx):
+        """Delete database/table"""
+
+    @mysql.command()
+    async def create(self, ctx, database_name):
+        """Create a database with MySQL"""
+        await ctx.trigger_typing()
+        self.cursorr.execute(f"CREATE DATABASE {database_name}")
```
Comment (aleclol) [inline sql/sql.py:75]: Has this been tested when you try to create a database with the name of an existing one?

Comment (aleclol) [inline sql/sql.py:75]: I believe this is blocking, you might want to run it in an executor

File: sql/sql.py [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+        await ctx.tick()
+        await ctx.send(f"{database_name} has been created.")
+
+    @mysql.command()
+    async def list(self, ctx):
+        """List databases in MySQL"""
+        await ctx.trigger_typing()
+        databases = ("show databases")
+        self.cursorr.execute(databases)
```
Comment (aleclol) [inline sql/sql.py:84]: Same thing about being blocking

File: sql/sql.py [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+        embed = discord.Embed(title="MySQL Databases", description="Here are the databases that your have on your machine:")
+        embed.add_field(name="Databases:", value=", ".join(x for (x,) in self.cursorr))
```
Comment (aleclol) [inline sql/sql.py:86]: If someone was too have too many databases, this could error for the embed character limit

File: sql/sql.py [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+        embed.set_footer(text="MySQL")
+        await ctx.send(embed=embed)
+
+    @delete.command(aliases=["database"])
+    async def db(self, ctx , database_name):
+        """Remove database"""
+        await ctx.trigger_typing()
+        self.cursorr.execute(f"DROP DATABASE IF EXISTS {database_name}")
```
Comment (aleclol) [inline sql/sql.py:94]: Same blocking thing

File: sql/sql.py [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+        await ctx.tick()
+        await ctx.send("If the database exists, it has been removed.")
```
Comment (aleclol) [inline sql/sql.py:96]: Maybe check if the database exists, and then delete it as someone may misspell something and believe they deleted it, when they never did

File: sql/sql.py [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+    @commands.is_owner()
+    @commands.group()
+    async def sql(self, ctx):
+        """SQL commands"""
+        pass
+
+    @sql.command()
+    async def version(self, ctx):
+        """Check what version of the SQL cog you have."""
+        await ctx.reply(f"This cog is on version {self.__version__}.", mention_author=False)
```
Comment (aleclol) [inline sql/sql.py:60]: You may want to use ctx.send as you used it in other commands, or switch the other ones to ctx.reply

**Review by aleclol — CHANGES_REQUESTED** [review]
Good Job

File: sql/sql.py [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+    def format_help_for_context(self, ctx):
+        """Thanks Sinbad!"""
+        pre_processed = super().format_help_for_context(ctx)
+        n = "\n" if "\n\n" not in pre_processed else ""
+        return f"{pre_processed}{n}\nCog Version: {self.__version__}"
+
+
+    async def red_delete_data_for_user(self, *, requester: RequestType, user_id: int) -> None:
+        # TODO: Replace this with the proper end user data removal handling.
+        super().red_delete_data_for_user(requester=requester, user_id=user_id)
```
Comment (Just-Jojo) [inline sql/sql.py:48]: This doesn't store data, just return here :)

File: sql/sql.py [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+        await ctx.reply(f"This cog is on version {self.__version__}.", mention_author=False)
+
+    @sql.group()
+    async def mysql(self, ctx):
+        """MySQL commands"""
+        pass
+
+    @mysql.group()
+    async def delete(self, ctx):
+        """Delete database/table"""
```
Comment (Just-Jojo) [inline sql/sql.py:69]: At the moment this only has one subcommand. If you don't plan on adding any more subcommands then I would suggest just making this `[p]sql mysql delete`

**Review by Just-Jojo — COMMENTED** [review]
Looks okay so far

File: sql/sql.py (hunk 2 above)
Comment (OofChair) [reply sql/sql.py]: Fixed in 4c40183

File: sql/sql.py (hunk 7 above)
Comment (OofChair) [reply sql/sql.py]: Fixed in e99249a

File: sql/sql.py (hunk 6 above)
Comment (OofChair) [reply sql/sql.py]: Fixed in e99249a

File: sql/sql.py (hunk 4 above)
Comment (OofChair) [reply sql/sql.py]: Should be fixed in 1bdd89d

File: sql/sql.py (hunk 5 above)
Comment (OofChair) [reply sql/sql.py]: Fixed in 3f64dee

File: sql/sql.py (hunk 3 above)
Comment (OofChair) [reply sql/sql.py]: Fixed in 3f64dee

File: sql/sql.py (hunk 2 above)
Comment (OofChair) [reply sql/sql.py]: Fixed in 3f64dee

**Comment by OofChair** [comment]
Seems all good, let's merge :D

**Pull request merged**